ui/components/data-sources/DataSourcesListItem.js

```diff
@@ -1,35 +1,68 @@
-import React from "react";
+import React, { Component } from "react";
+
 import {
     ListView,
     DropdownKebab,
     MenuItem
 } from "patternfly-react";
 
-const DataSourcesListItem = ({ item }) => {
-    const { type, name } = item;
-
-    return (
-        <ListView.Item
-            className="ds-list-item"
-            heading={type}
-            description="---"
-            leftContent={(
-                <span className="list-item-name">{name}</span>
-            )}
-            actions={(
-                <div>
-                    <DropdownKebab id="data-source-list-item-dropdown" pullRight>
-                        <MenuItem>
-                            Edit Data Source
-                        </MenuItem>
-                        <MenuItem>
-                            Delete Data Source
-                        </MenuItem>
-                    </DropdownKebab>
-                </div>
-            )}
-        />
-    );
-};
-
-export { DataSourcesListItem };
+import { graphql } from "react-apollo";
+import DeleteDataSource from "../../graphql/DeleteDataSource.graphql";
+import GetDataSources from "../../graphql/GetDataSources.graphql";
+
+class DataSourcesListItem extends Component {
+
+    constructor(props) { //eslint-disable-line
+        super(props);
+    }
+
+    removeOneDatasource(dataSourceId) {
+        const self = this;
+        self.props.mutate({
+            variables: { dataSourceId },
+            refetchQueries: [{
+                query: GetDataSources,
+                variables: { name: "" }
+            }]
+        }).then(() => {
+            self.props.onDelete();
+        });
+    }
+
+    render() {
+        const self = this;
+        const { type, name, id } = self.props.item;
+        return (
+            <ListView.Item
+                className="ds-list-item"
+                heading={type}
+                leftContent={(
+                    <span className="list-item-name">{name}</span>
+                )}
+                actions={(
+                    <div>
+                        <DropdownKebab id="data-source-list-item-dropdown" pullRight>
+                            <MenuItem>
+                                Edit Data Source
+                            </MenuItem>
+                            <MenuItem
+                                eventKey={2}
+                                onSelect={() => {
+                                    if (window.confirm("Are you sure you wish to delete this item?")) {
```

Comment:
We should use a patternfly confirmation box here: https://rawgit.com/patternfly/patternfly-react/gh-pages/index.html?selectedKind=patternfly-react%2FCommunication%2FMessage%20Dialog&selectedStory=Question%20Dialog%20Example&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybooks%2Fstorybook-addon-knobs

Reply:
Agree, this component shouldn't contain specific graphql logic (otherwise edit should be here as well). Let's add a new component in `DataSourcesContainer` that renders patternfly's `MessageDialog` and add this logic inside.
You can take a look at how's implemented in #29 (the same but with a Modal):
[in DataSourcesList](https://github.com/aerogear/data-sync-ui/pull/29/files#diff-e2f0f644d847f64f715e7317b123ca5dR21)
[in DataSourcesListItem](https://github.com/aerogear/data-sync-ui/pull/29/files#diff-8875867d8c03c06a146dba48848066efR22)
Important also not to change the export mode [data-sources/index](https://github.com/aerogear/data-sync-ui/pull/29/files#diff-7a96b4fb5f8b8997dd97aa0e10b08b90R2)